src/main/java/tasks/Task.java

```diff
@@ -0,0 +1,61 @@
+package tasks;
+import java.io.Serializable;
+import java.time.LocalDate;
+/**
+ * Representation of a task
+ * recorded by the chatbot.
+ * @author Alvis Ng (supermii2)
+ */
+public abstract class Task implements Serializable {
+    /** Name of task */
+    protected String taskName;
+    /** One letter abbreviation for the task. */
+    protected String oneLetterAbbrev;
+    /** Whether the task has been completed */
+    private boolean completed;
```

Comment:
You can consider renaming your boolean variable to `isCompleted`. Just a small nitpick!

Reply:
Thank you! Good catch